modules/nf-core/nonpareil/nonpareilcurvesr/main.nf

```diff
@@ -0,0 +1,57 @@
+process NONPAREIL_NONPAREILCURVESR {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/nonpareil:3.5.5--r43hdcf5f25_0':
+        'biocontainers/nonpareil:3.5.5--r43hdcf5f25_0' }"
+
+    input:
+    tuple val(meta), path(npos)
+
+    output:
+    tuple val(meta), path("*.json"), emit: json, optional: true
+    tuple val(meta), path("*.tsv" ), emit: tsv , optional: true
+    tuple val(meta), path("*.csv" ), emit: csv , optional: true
+    tuple val(meta), path("*.pdf" ), emit: pdf , optional: true
+
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    NonpareilCurves.R \\
+        $args \\
+        --json ${prefix}.json \\
+        --tsv ${prefix}.tsv \\
+        --csv ${prefix}.csv \\
+        --pdf ${prefix}.pdf \\
```

Comment:
Nice! Is this the proper way to define the optional outputs instead of defining a function for output options?

Reply:
Oh good point... I set them optional initially as they technically are, but I realised they are all small so could just generate alll...

Do you have a suggestion how we could ask the user/pipeline dev which ones they want?

Reply:
Emm, I am not sure if I could come up a wiser way. How about something like this, only keep the `tsv` output? I guess we could also keep all of them since they are quite small anyway. 
```
    output:
    tuple val(meta), path("**.{json,tsv,csv,pdf}"), emit: json, optional: true

    """
    NonpareilCurves.R \\
        $args \\
        --tsv ${prefix}.tsv \\
        $npos
    """

```

Reply:
The problem there is it's more tricky to 'pick one' for downstream processing (e.g. multiqc), but still publish the others...

I don't want to force devs  to pass output flags via `args`... as this could become confusing (if you see what I mean)

Reply:
Something like [this](https://github.com/nf-core/modules/blob/16af6ca4aeb31dcbb2d652ada5f395d62793426e/modules/nf-core/blast/blastdbcmd/main.nf#L32), could not work?

Reply:
Mmm, issue there is that is a fixed thing, whereas in this case I have to specify the file name after... `--tsv <filename> --csv <filename>` etc...

Reply:
and I don't want pipeline devs having to specify `--tsv ${prefix}.tsv` in their modules config etc.